src/shell-app.c

```diff
@@ -937,7 +939,10 @@ shell_app_sync_running_state (ShellApp *app)
 
   if (app->state != SHELL_APP_STATE_STARTING)
     {
-      if (app->running_state->interesting_windows == 0)
+      /* We check if the startup completed as a workaround for apps that show
+       * a window and quickly remove it before the main window shows up, which
+       * is the case with Maps. */
+      if (app->running_state->interesting_windows == 0 && app->running_state->startup_complete)
```

Comment:
I wonder if instead of tracking the startup completion this way here you couldn't check that there are no speedwagon windows in addition to interesting windows.
I.e. (pseudo-code) ` if (app->running_state->interesting_windows == 0 && app_has_no_speedwagon_windows(app))`